### PR TITLE
Create typings for the interop API

### PR DESCRIPTION
**Pull request by Feiyang1**

### CODE REVIEW

File: scripts/exp/create-overloads.ts [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as yargs from 'yargs';
+import * as ts from 'typescript';
+import * as fs from 'fs';
+
+/**
+ * TODO:
```
Comment (hsubox76) [inline scripts/exp/create-overloads.ts:23]: This seems to be done.

File: common/api-review/auth-exp.api.md [hunk 2]
```diff
@@ -184,6 +185,16 @@ export const debugErrorMap: AuthErrorMap;
 // @public
 export function deleteUser(user: User): Promise<void>;
 
+// @public (undocumented)
```
Comment (hsubox76) [inline common/api-review/auth-exp.api.md:188]: Needs documentation now that it's publicly exposed?

File: scripts/exp/create-overloads.ts [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+        ts.setTextRange(node, {
+          pos: node.getStart(sourceFile),
+          end: node.getEnd()
+        });
+
+        overloads.push(
+          ts.updateFunctionDeclaration(
+            node,
+            node.decorators,
+            [],
```
Comment (hsubox76) [inline scripts/exp/create-overloads.ts:182]: Deliberately removing any modifiers? What are modifiers? "public|private|protected"?

**Review by hsubox76 — APPROVED** [review]
LG, just a couple small things and a question for my benefit.

File: common/api-review/auth-exp.api.md (hunk 2 above)
Comment (Feiyang1) [reply common/api-review/auth-exp.api.md:188]: Yes, we should do a sweep for all packages at some point and add any missing documentation. I will leave this PR as is for now.

File: scripts/exp/create-overloads.ts (hunk 3 above)
Comment (Feiyang1) [reply scripts/exp/create-overloads.ts:182]: Yes, to remove `export` and `declare` which are not needed in the `declare module` block

**Pull request merged**